orchagent/routeorch.cpp

```diff
@@ -1527,11 +1545,19 @@ bool RouteOrch::removeNextHopGroup(const NextHopGroupKey &nexthops)
 
     m_nextHopGroupCount--;
     gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_NEXTHOP_GROUP);
+    MuxOrch* mux_orch = gDirectory.get<MuxOrch*>();
+    sai_object_id_t mux_tunnel_nh_id = mux_orch->getTunnelNextHopId();
 
     set<NextHopKey> next_hop_set = nexthops.getNextHops();
     for (auto it : next_hop_set)
     {
-        m_neighOrch->decreaseNextHopRefCount(it);
+        // skip ref count update if the underlying nh from mux is tunnel nh
```

Comment:
Should the interface status be checked here as well? if the interface is down (and the refcount is decreased in the interface down path) we might decrease the refcount twice